### DISCUSSION TITLE
Having problem connecting to SQL Server 2022.

### DISCUSSION THREAD

**Discussion by TomMiller-mas** [tediousjs/tedious#1582] [Q&A]
Does v16 support SQL Server 2022? 
SQL Server 2022 no longer installs the native driver.
Does TDS have a native connection directly to SQL Server or is it using one of the drivers?

node 16.17.1
mssql 10.0.1 (uses v16 of tedious)
SQL Server Express 2022
Windows 11

**Comment by TomMiller-mas** [comment]
I have done several tests of uninstalling 2022 and installing 2019 getting it to work, then reinstalling 2022.  There is definitely an issue with SQL Server 2022 support.  I also installed the native client and that didn't make any difference. Just an FYI, the installer is no longer setting the default port for the server to 1433.  You have to do that yourself. 

Has any one gotten tedious to work with any version of SQL Server 2022?

**Reply by arthurschreiber** [reply]
Tedious does not rely on the native client libraries, it's a fully standalone implementation of the TDS protocol used by sql server.

SQL Server 2022 is also fully supported. Do you have more information about your setup and the failures you are encountering?

Do you have tcp connections enabled in SQL Server 2022?

**Reply by TomMiller-mas** [reply]
Windows 11
New install of SQL Server Express 2022
node.js 16.17.1
Express.js 4.x
npm mssql 10.0.1 (using Tedious 16.x)

The first connection string works fine with the same setup and SQL Server Dev 2019 connection through our test system. I also setup SQL Server 2019 Express and it worked also. Both servers have TCP/IP turned on and the port set manually to 1433 on both 2019 systems and 2022.

**Server=MyLaptop\SQLEXPRESS,1433;Database=MyTestDatabase;
    User Id=xxxxxxxx;Password=yzyzyzyzyzyzyzyzyzyzyzyzyz;Encrypt=false;TrustServerCertificate=true**

Other ones that I have tried.

Server=MyLaptop;Database=MyTestDatabase;
    User Id=xxxxxxxx;Password=yzyzyzyzyzyzyzyzyzyzyzyzyz;Encrypt=false;TrustServerCertificate=true

Server=localhost;Database=MyTestDatabase;
    User Id=xxxxxxxx;Password=yzyzyzyzyzyzyzyzyzyzyzyzyz;Encrypt=false;TrustServerCertificate=true

Server=localhost\SQLEXPRESS;Database=MyTestDatabase;
    User Id=xxxxxxxx;Password=yzyzyzyzyzyzyzyzyzyzyzyzyz;Encrypt=false;TrustServerCertificate=true

Server=localhost\SQLEXPRESS, 1433;Database=MyTestDatabase;
    User Id=xxxxxxxx;Password=yzyzyzyzyzyzyzyzyzyzyzyzyz;Encrypt=false;TrustServerCertificate=true

==================================================
I am going to test using Tedious directly tomorrow.  I just wanted to make sure I didn't have to worry about drivers before testing.  Is there anything different about connecting to 2022 vs 2019? Something I need to take out of my connection string or is a new requirement?   
==================================================
**Error:**

{
  "code": "ETIMEOUT",
  "originalError": {
    "code": "ETIMEOUT"
  },
  "name": "ConnectionError"
}

**Reply by TomMiller-mas** [reply]
I got it working.  It seemed mostly to be about casing.  I swear I am using the same stuff as before when it wouldn't work. Maybe it was the trustServerCertificate casing or something.  Sorry to bother.

**Comment by TomMiller-mas** [comment]
These both worked with SQL Server 2022

**Tedious**
var config = {
  server: "127.0.0.1", // or "localhost"
  options: {
    trustServerCertificate: true,
    database: "MyDatabase",
  },
  authentication: {
    type: "default",
    options: {
      userName: "xxxxxxxxx",
      password: "yzyzyzyzzyzyzyzyzyzzyzyzyzyzy",
    },
  },
};

**mssql libary connection string** 

        if (data[k].port != "") {dbPort = "," + data[k].port;}
        var mssqlConnStr =
          "Server=" +data[k].server +
          dbPort +
          ";Database=" +
          data[k].database +
          ";User Id=" +data[k].usrNm +
          ";Password=" + data[k].usrPwd +
          ";encrypt=false;trustServerCertificate=true";
